src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -1264,26 +1313,27 @@ private void AggregateRenderPackages(IEnumerable<HelixRenderPackage> packages)
                     // with `points`, `lines`, or `mesh`. For each RenderPackage, we check whether the geometry dictionary
                     // has entries for the points, lines, or mesh already. If so, we add the RenderPackage's geometry
                     // to those geometry objects.
-                    
+
                     var baseId = rp.Description;
                     if (baseId.IndexOf(":", StringComparison.Ordinal) > 0)
                     {
                         baseId = baseId.Split(':')[0];
                     }
-                    var id = baseId;
+                    
                     //If this render package belongs to special render package, then create
                     //and update the corresponding GeometryModel. Sepcial renderpackage are
                     //defined based on its description containing one of the constants from
                     //RenderDescriptions struct.
-                    if (UpdateGeometryModelForSpecialRenderPackage(rp, id))
+                    if (UpdateGeometryModelForSpecialRenderPackage(rp, baseId))
                         continue;
 
                     var drawDead = InCustomNode() && !customNodeIdents.Contains(baseId);
 
+                    string id;
                     var p = rp.Points;
                     if (p.Positions.Any())
                     {
-                        id = baseId + PointsKey;
+                        id = rp.Description + PointsKey;
```

Comment:
it needs to store points separately for each array item, so `var_guid:0:1` is used instead of `var_guid`